data/campaigns/The_Hammer_of_Thursagan/scenarios/12_The_Underlevels.cfg

```diff
@@ -291,6 +337,33 @@
             speaker=Aiglondur
             message= _ "Aye. Karrag, calling his troops to battle. Only the Dark Gods know what hellspawn the lich will summon. AXES UP!"
         [/message]
+        
+        [store_unit]
+            [filter]
+                side=1
+            [/filter]
+            variable=units
+        [/store_unit]
+        [for]
+            array=units
+            [do]
+                [object]
+                    silent=yes
+                    take_only_once=yes
+                    [filter]
+                        id=$units[$i].id
+                    [/filter]
+                    [effect]
+                        apply_to=vision_costs
+                        replace=yes
+                        [vision_costs]
+                            impassable=3
+                        [/vision_costs]
+                    [/effect]   
+                [/object]
+            [/do]
+        [/for]
+        {CLEAR_VARIABLE units}
```

Comment:
If already applying this at recruited and recalled units, it is eventually not needed here. In and case, a unit placed event covers all.